pyresample/geometry.py

```diff
@@ -1026,6 +1087,31 @@ def freeze(self, lonslats=None, resolution=None, shape=None, proj_info=None):
           the shape of the resulting area.
         proj_info:
           complementing parameters to the projection info.
+        antimeridian_mode:
+            How to handle lon/lat data crossing the anti-meridian of the
+            projection. This currently only effects lon/lat geographic
+            projections and data cases not covering the north or south pole.
+            The possible options are:
+
+            * modify_extents: Set the X bounds to the edges of the data, but
+                add 360 to the right-most bound. This has the effect of making
+                the area coordinates continuous from the left side to the
+                right side. However, this means that some coordinates will be
+                outside the coordinate space of the projection. Although most
+                PROJ and pyresample functionality can handle this there may be
+                some edge cases.
+            * modify_projection: Change the prime meridian of the projection
+                from 0 degrees longitude to 180 degrees longitude. This has
+                the effect of putting the data on a continuous coordinate
+                system. However, this means that comparing data resampled to
+                this resulting area and an area not over the anti-meridian
+                would be more difficult.
+            * global_extents: Ignore the bounds of the data and use -180/180
+                degrees as the west and east bounds of the data. This will
+                generate a large output area, but with the benefit of keeping
+                the data on the original projection. Note that some resampling
+                methods may produce artifacts when resampling on the edge of
+                the area (the anti-meridian).
```

Comment:
@mraspaud @pnuu @gerritholl I think the last remaining question is what are the best names for these parameters. In another comment Panu pointed out that `global_extents` is the only one that doesn't have an imperative verb at the front. He also suggested using "shift_prime_meridian" instead of "modify_projection", but my worry with that is that in the future modifying the projection to support the data may not only include shifting the prime meridian.

Thoughts? Opinions?

Reply:
`make_extent_global` or `globalize_extent` would be an imperative form of `global_extents`.  But it's not actually global, is it?  It covers all longitudes, but must not necessarily cover all latitudes.  It anyway should affect only the longitude extent, not the latitude extent, shouldn't it?  The docstring in lines 1091 and 1092 mentions lon/lat, but this should be only about lon?

About `modify_projection`: it would seem that it's not accurate to speak of a projection at all here.  When we describe data in lat/lon then the data are unprojected, or are they?

Reply:
I would replace `projection` by `CRS` and change `modify_projection` to `modify_CRS`.  I can see how a future case of, for example, data covering a pole may modify lat/lons by putting the pole somewhere different, yet retain spherical coordinates therefore still not being technically a projection (but rather a confusion ☺).

Reply:
The poles are a coordinate singularity, so lat/lon coordinates are simply not suitable around the poles.

Reply:
I like `modify_crs` (lower case, to be more pythonic). And indeed, I was only thinking about the current case of adjusting the prime meridian.

Similarly `global_extents` would be future proof for any projection type, although what @gerritholl sait about the actual globality is a valid point. Some (hasty!) suggestions for names for this: `use_full_width(_extents)`, `wrap_x_extents`, `cover_all_meridians`.

Reply:
The thing about the "global_extents" not being global is a little odd. Right now it is global in the longitudinal dimension, but not necessarily in latitude. I would hope users wouldn't be too confused that a parameter dealing with the anti-meridian would only effect meridians and not parallels. We could call it `expand_extents` which can give the general impression of "we're going to increase the range of the extents to fit more than just the data".

Reply:
`ignore_extents` instead of `global_extents`? and `modify_crs` sounds good.

Reply:
Why "ignore"? What is being ignored? Oh you mean ignoring the extents of the data...eh I'm not sure I like that. I'll change to `modify_crs` for that one, but still not sure on extents.